contributed_definitions/nyaml/NXxrd_pan.yaml

```diff
@@ -146,42 +146,75 @@ NXxrd_pan(NXxrd):
           rank: 1
           dim: (nDet,)
       two_theta(NX_FLOAT):
+        unit: NX_ANGLE
         doc: |
           Axis scale represeting the 2-theta range of the difractogram.
         dimensions:
           rank: 1
           dim: (nDet,)
       omega(NX_FLOAT):
         exists: optional
+        unit: NX_ANGLE
         doc: |
           The omega range of the difractogram.
         dimensions:
           rank: 1
           dim: (nDet,)
       phi(NX_FLOAT):
         exists: optional
+        unit: NX_ANGLE
         doc: |
-          The phi range of the difractogram
+          The phi range of the diffractogram.
         dimensions:
           rank: 1
           dim: (nDet,)
       chi(NX_FLOAT):
         exists: optional
+        unit: NX_ANGLE
         doc: |
           The chi range of the difractogram
         dimensions:
           rank: 1
           dim: (nDet,)
-    q_plot(NXdata):
+      q_parallel(NX_FLOAT):
+        exists: optional
+        unit: NX_ANY
+        doc: |
+          The scattering vector component parallel to the sample surface.
+      q_perpendicular(NX_FLOAT):
+        exists: optional
+        unit: NX_ANY
+        doc: |
+          The scattering vector component perpendicular to the sample surface.
+      q_norm(NX_FLOAT):
+        exists: optional
+        unit: NX_ANY
+        doc: |
+          The norm of scattering vector q.
+    q_data(NXdata):
       exists: optional
       doc: |
         Desired plot would be q_vector vs intensity.
       q(NX_FLOAT):
         doc: |
-          Axis scale representing wavevector for scatter energy.
+          Scattering vector link to the 'q_norm' of experiment_result.
```

Comment:
It's a bit unclear what you mean here.

Generally, links cannot be enforced in Nexus, so you should also add a description for this value and then explain that it should be linked to `q_norm`. You can also state that these two are the same concept (unfortunately only in text form yet).

Reply:
Noted!